syntax = "proto3";

package contacts;

service ContactService {
  rpc AddContact (Contact) returns (AddContactResponse);
  rpc GetContact (ContactRequest) returns (Contact);
  rpc ListContacts (Empty) returns (ContactList);
}

message Empty {}

enum PhoneType {
  MOBILE = 0;
  HOME = 1;
  WORK = 2;
}

enum Category {
  FAMILY = 0;
  PERSONAL = 1;
  BUSINESS = 2;
}

message PhoneNumber {
  string number = 1;
  PhoneType type = 2;
}

message Contact {
  string name = 1;
  repeated PhoneNumber phones = 2;
  Category category = 3;
}

message ContactRequest {
  string name = 1;
}

message ContactList {
  repeated Contact contacts = 1;
}

message AddContactResponse {
  string message = 1;
}
